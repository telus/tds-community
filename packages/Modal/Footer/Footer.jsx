import React from 'react'
import PropTypes from 'prop-types'
import Button from '@tds/core-button'
import Box from '@tds/core-box'
import Link from '@tds/core-link'
import { Delete } from '@tds/core-interactive-icon'

import {
  BUTTON_RANKS,
  CANCELLATION_BUTTON_TYPES,
  BUTTON_VARIANTS,
  LINK_ICON_POSITIONS,
} from '../configs'
import { LinkWrapper } from '../styles'

const CancellationButton = (handleClose, buttonType, text) => {
  switch (buttonType) {
    case CANCELLATION_BUTTON_TYPES.LINK_WITH_ICON:
      return (
        <LinkWrapper>
          <Link
            onClick={handleClose}
            icon={Delete}
            iconPosition={LINK_ICON_POSITIONS.RIGHT}
            data-testid="cancel-link"
          >
            {text}
          </Link>
        </LinkWrapper>
      )
    case CANCELLATION_BUTTON_TYPES.LINK_WITHOUT_ICON:
      return (
        <LinkWrapper>
          <Link onClick={handleClose} data-testid="cancel-link">
            {text}
          </Link>
        </LinkWrapper>
      )
    default:
      return (
        <Button variant="danger" onClick={handleClose} data-testid="cancel-button">
          {text}
        </Button>
      )
  }
}

const Footer = ({
  cancelButtonText,
  cancelButtonType,
  confirmButtonText,
  confirmButtonVariant,
  handleClose,
  handleConfirm,
}) => {
  return (
    <Box between={3} inline={{ xs: false, md: true }}>
      <Button
        variant={confirmButtonVariant}
        rank={BUTTON_RANKS.MAIN}
        onClick={handleConfirm}
        data-testid="confirm-button"
      >
        {confirmButtonText}
      </Button>
      {cancelButtonText && CancellationButton(handleClose, cancelButtonType, cancelButtonText)}
    </Box>
  )
}

Footer.propTypes = {
  cancelButtonText: PropTypes.string,
  cancelButtonType: PropTypes.oneOf(Object.values(CANCELLATION_BUTTON_TYPES)),
  confirmButtonText: PropTypes.string.isRequired,
  confirmButtonVariant: PropTypes.oneOf(Object.values(BUTTON_VARIANTS)),
  handleClose: PropTypes.func.isRequired,
  handleConfirm: PropTypes.func.isRequired,
}

Footer.defaultProps = {
  cancelButtonText: '',
  cancelButtonType: CANCELLATION_BUTTON_TYPES.LINK_WITHOUT_ICON,
  confirmButtonVariant: BUTTON_VARIANTS.STANDARD,
}

export default Footer
