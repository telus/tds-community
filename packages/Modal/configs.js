export const HEADER_LEVELS = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
}

export const TEXT_SIZES = {
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
}

export const BUTTON_RANKS = {
  MAIN: 'main',
  COMMON: 'common',
}

export const CANCELLATION_BUTTON_TYPES = {
  BUTTON: 'button',
  LINK_WITH_ICON: 'linkWithIcon',
  LINK_WITHOUT_ICON: 'linkWithoutIcon',
}

export const BUTTON_VARIANTS = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  INVERTED: 'inverted',
  STANDARD: 'standard',
  BRAND: 'brand',
  DANGER: 'danger',
}

export const NOTIFICATION_VARIANTS = {
  INSTRUCTIONAL: 'instructional',
  BRANDED: 'branded',
  SUCCESS: 'success',
  ERROR: 'error',
  WARNING: 'warning',
}

export const NOTIFICATION_COPY = {
  EN: 'en',
  FR: 'fr',
}

export const LINK_ICON_POSITIONS = {
  LEFT: 'left',
  RIGHT: 'right',
}
