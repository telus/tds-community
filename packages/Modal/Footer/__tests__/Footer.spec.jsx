import React from 'react'
import { mount } from 'enzyme'

import Footer from '../Footer'
import { CANCELLATION_BUTTON_TYPES, BUTTON_VARIANTS } from '../../configs'

describe('Footer', () => {
  const handleClose = jest.fn()
  const handleConfirm = jest.fn()

  const doMount = (props = {}) =>
    mount(
      <Footer
        cancelButtonText="Cancel"
        cancelButtonType={CANCELLATION_BUTTON_TYPES.LINK_WITHOUT_ICON}
        confirmButtonText="Confirm"
        confirmButtonVariant={BUTTON_VARIANTS.STANDARD}
        handleClose={handleClose}
        handleConfirm={handleConfirm}
        {...props}
      />
    )

  it('renders', () => {
    const footer = doMount()

    expect(footer).toMatchSnapshot()
  })
})
