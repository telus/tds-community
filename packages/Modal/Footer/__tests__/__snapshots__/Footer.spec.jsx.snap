// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
.c1 {
  border-width: 0;
  border-radius: 4px;
  word-wrap: break-word;
  font-size: 1rem;
  -webkit-letter-spacing: -0.8px;
  -moz-letter-spacing: -0.8px;
  -ms-letter-spacing: -0.8px;
  letter-spacing: -0.8px;
  line-height: 1.5;
  word-wrap: break-word;
  font-weight: 700;
  font-family: "TELUS-Web","Helvetica Neue",Helvetica,Arial,sans-serif;
  margin: 0;
  padding: 0 2rem;
  cursor: pointer;
  background: none;
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  min-height: 3.25rem;
  background-color: #2B8000;
  color: #fff;
  -webkit-transition: background 0.2s,color 0.2s,border 0.2s ease;
  transition: background 0.2s,color 0.2s,border 0.2s ease;
}

.c1:after {
  content: "";
  min-height: inherit;
  font-size: 0;
}

.c1:hover {
  background-color: #1F5C09;
  box-shadow: 0 0 0 0.125rem #1F5C09;
}

.c1:active {
  background-color: #163E06 !important;
}

.c1:focus {
  outline: none !important;
  background-color: #1F5C09;
  box-shadow: 0 0 0 0.1875rem #509F33,0 0 0 0.125rem #fff inset;
}

.c2 {
  width: 100%;
  margin-top: -1px;
}

@media (min-width:768px) {
  .c1 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    width: auto;
    min-width: 180px;
  }
}

@media (hover:none) {
  .c1:hover {
    box-shadow: none;
    background-color: #2B8000;
    color: #fff;
  }
}

@media (prefers-reduced-motion:reduce) {
  .c1 {
    -webkit-transition: none !important;
    transition: none !important;
  }
}

@media (max-width:767px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c0 > *:not(:last-child) {
    margin-bottom: 1rem;
  }
}

@media (min-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c0 > *:not(:last-child) {
    margin-right: 1rem;
  }
}

<Footer
  cancelButtonText=""
  cancelButtonType="linkWithoutIcon"
  confirmButtonVariant="standard"
  footerContent={
    Object {
      "cancelCTAContent": Object {
        "showIcon": true,
        "text": "Cancel",
        "type": "button",
        "variant": "danger",
      },
      "confirmCTAContent": Object {
        "rank": "main",
        "text": "Confirm",
        "variant": "standard",
      },
    }
  }
>
  <Box
    between={3}
    inline={
      Object {
        "md": true,
        "xs": false,
      }
    }
    tag="div"
  >
    <styled.div
      between={3}
      inline={
        Object {
          "md": true,
          "xs": false,
        }
      }
      tag="div"
    >
      <StyledComponent
        between={3}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
            ],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        inline={
          Object {
            "md": true,
            "xs": false,
          }
        }
        tag="div"
      >
        <div
          className="c0"
        >
          <Button
            data-testid="confirm-button"
            rank="main"
            type="button"
            variant="standard"
          >
            <Button__StyledButton
              data-testid="confirm-button"
              rank="main"
              type="button"
              variant="standard"
            >
              <StyledComponent
                data-testid="confirm-button"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "Button__StyledButton-sa7xo3-0",
                      "isStatic": false,
                      "lastClassName": "c1",
                      "rules": Array [
                        "border-width: 0;",
                        "border-radius: 4px;",
                        "word-wrap: break-word;",
                        "font-size: 1rem;",
                        "letter-spacing: -0.8px;",
                        "line-height: 1.5;",
                        "word-wrap: break-word;",
                        "font-weight: 700;",
                        "font-family: \\"TELUS-Web\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif;",
                        "margin: 0;",
                        "padding: 0 2rem;",
                        "cursor: pointer;",
                        "background: none;",
                        "transition: background 0.2s;",
                        "display: flex;",
                        "width: 100%;",
                        "align-items: center;",
                        "justify-content: center;",
                        "text-align: center;",
                        "min-height: 3.25rem;",
                        "@media (min-width: 768px) {",
                        "display: inline-flex;",
                        "width: auto;",
                        "min-width: 180px;",
                        "}",
                        "&:after {",
                        "content: \\"\\";",
                        "min-height: inherit;",
                        "font-size: 0;",
                        "}",
                        [Function],
                      ],
                    },
                    "displayName": "Button__StyledButton",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "Button__StyledButton-sa7xo3-0",
                    "target": "button",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                rank="main"
                type="button"
                variant="standard"
              >
                <button
                  className="c1"
                  data-testid="confirm-button"
                  type="button"
                >
                  <Button__ButtonTextWrapper
                    isOldButton={false}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Button__ButtonTextWrapper-sa7xo3-1",
                            "isStatic": false,
                            "lastClassName": "c2",
                            "rules": Array [
                              [Function],
                            ],
                          },
                          "displayName": "Button__ButtonTextWrapper",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Button__ButtonTextWrapper-sa7xo3-1",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      isOldButton={false}
                    >
                      <span
                        className="c2"
                      />
                    </StyledComponent>
                  </Button__ButtonTextWrapper>
                </button>
              </StyledComponent>
            </Button__StyledButton>
          </Button>
        </div>
      </StyledComponent>
    </styled.div>
  </Box>
</Footer>
`;
